### PR TITLE
WIP: Create Manage Local Users in Studio for Tencent

### PR DESCRIPTION
**Pull request by MarkvanMents**

### CODE REVIEW

**Comment by mengdix** [comment]
By Studio Team's request, can we change the filename from "tencent-local-users-in-studio.md" to "local-users-in-studio.md" ?

**Comment by MarkvanMents** [comment]
Hi Meng, @mengdix 
I added "Tencent" to the front so that I can easily find all the documents that are related to tencent as they will be filed together.
However, I'll change the name for you:
We can discuss best practices later.

**Comment by MarkvanMents** [comment]
@MariaShaposhnikova - adding you for information as this is related a bit to Studio

**Comment by MarkvanMents** [comment]
@mengdix 
Hi Meng,
This document still needs to be worked on - do we need to do something so that it supports yesterday's release, or is this for later?
Thanks
Mark

File: content/developerportal/deploy/managing-local-users-after-deployment.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+With Mendix Studio Pro, you can configure an Administrator account directly in your security settings. See [Administrator](/refguide8/administrator) for more information.
+
+With Mendix Studio on Tencent Cloud, it’s also possible to directly publish your app to your deployment environment and manage your app’s users locally from there. For your convenience, we’ve included the “Administration” module in the starter app template to make that happen. This article guides you through accessing this and managing users.
+
+## 2 Accessing the Administrator Account
+
+For each app published from Mendix Studio on Tencent Cloud, a default administrator user is created.
+
+### 2.1 Administrator Properties
```
Comment (MarkvanMents) [inline content/developerportal/deploy/managing-local-users-after-deployment.md:31]: This whole section is just a cut and paste from https://docs.mendix.com/refguide8/administrator.
Is there any reason why we need it here and can't just link to the original document?

File: content/developerportal/deploy/managing-local-users-after-deployment.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+    ![](attachments/managing-local-users-after-deployment/apply-changes.png)
+
+## 3 Managing App Users
+
+Using your freshly updated login credentials, you are ready to start managing your users. The *Administration* module is included in every new blank app. This section describes how to log into your newly published app and visit the **User Administration** section.
+
+### 3.1 Introduction
+
+The Administration module is shipped with all Mendix Project Themes that can be used for new App creation. The module contains administration functionalities to manage App Users and to see App statistics like runtime information, sessions and schedules events.
```
Comment (MarkvanMents) [inline content/developerportal/deploy/managing-local-users-after-deployment.md:101]: We seem to say several times that the administration module is in the app - do we need to say it here as well?

File: content/developerportal/deploy/managing-local-users-after-deployment.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+2. Log in with the [default Admin username](/refguide8/administrator#user-name) and your [newly changed Admin Password](/refguide8/administrator#password).
+
+    ![](attachments/managing-local-users-after-deployment/sign-in.png)
+
+3. Select the **User Management** option from your App’s navigation menu. You will see this option because you are logged in as administrator.
+
+    ![](attachments/managing-local-users-after-deployment/account-overview.png)
+
+{{% alert type="info" %}}
+If you created your app using an older version of Mendix, this module might not be available to you. Please [consult the How-to guide on importing modules](/howto8/integration/importing-and-exporting-objects) into your project and redeploy.
```
Comment (MarkvanMents) [inline content/developerportal/deploy/managing-local-users-after-deployment.md:127]: Does this give the user enough information on what to do? What is the module called and can they import it through Marketplace, or do they need to get it from somewhere else?

File: content/developerportal/deploy/managing-local-users-after-deployment.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+![](attachments/managing-local-users-after-deployment/edit-existing.png)
+
+Change the user properties where necessary and click **Save**.
+
+![](attachments/managing-local-users-after-deployment/local-user-details.png)
+
+#### 3.3.3 Creating or Editing Web Service Users
+
+The User Management Overview also allows you to create *Web Service Users*. These connect your Mendix App with other (Mendix) apps [when publishing Web Services](/refguide/published-web-services). In some cases, to access services, other apps may need to authenticate themselves with your Mendix App. You can use this option to create dedicated accounts for each user of your Mendix App’s *Web Services*.
```
Comment (MarkvanMents) [inline content/developerportal/deploy/managing-local-users-after-deployment.md:156]: I find this very unclear. Is there a better way of explaining what web service users are? Where are their credentials held, for example.

File: content/developerportal/deploy/managing-local-users-after-deployment.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+![](attachments/managing-local-users-after-deployment/new-web-service-user.png)
+
+Simply select **New web service user** in the User Management Overview and fill out the required fields.
```
Comment (MarkvanMents) [inline content/developerportal/deploy/managing-local-users-after-deployment.md:160]: This seems to be the same information needed for local users. How are service credentials synced between apps. Or is that not the purpose?

**Review by MarkvanMents — COMMENTED** [review]
@mengdix 
Hi Meng, 
I have reformatted the document and made a few textual changes to be consistent with other documents. If you want to see the difference between the Paper doc and this, you can look at this commit: https://github.com/mendix/docs/pull/3412/commits/399979e4fccd678c7299e4fe8c12b783f4086189

I have added a few questions to the text, let me know if you need any more information.

If this is not completed tomorrow, you can always get one of my colleagues to publish it when it is ready.

Mark

File: content/developerportal/deploy/managing-local-users-after-deployment.md (hunk 1 above)
Comment (mengdix) [reply content/developerportal/deploy/managing-local-users-after-deployment.md:31]: For ease of use, not having to jump to another page simply to understand the default username and role, I have taken it from another doc, but some parts are worded specifically for Tencent (the main doc refers to other deployment scenarios)

File: content/developerportal/deploy/managing-local-users-after-deployment.md (hunk 3 above)
Comment (mengdix) [reply content/developerportal/deploy/managing-local-users-after-deployment.md:127]: They'll be able to get it from App Store in Studio Pro. It's referred to as "Administration" module. Perhaps we can substitute "this" for "the Administration module", for the sake of clarity.

File: content/developerportal/deploy/managing-local-users-after-deployment.md (hunk 4 above)
Comment (mengdix) [reply content/developerportal/deploy/managing-local-users-after-deployment.md:156]: Basically you can create separate credentials for others (humans or apps) to access your app's published web services. It's like creating API users in Google Cloud console, etc. I recommend we keep it basic for now, and if people give feedback requiring more detail on this, perhaps flag with the SP team to create a specific page explaining security settings for web service users.

File: content/developerportal/deploy/managing-local-users-after-deployment.md (hunk 5 above)
Comment (mengdix) [reply content/developerportal/deploy/managing-local-users-after-deployment.md:160]: That is up to the developer, but that person would have to share the credentials with the other app some way.

File: content/developerportal/deploy/managing-local-users-after-deployment.md (hunk 1 above)
Comment (MarkvanMents) [reply content/developerportal/deploy/managing-local-users-after-deployment.md:31]: OK, I'll leave it in then.

File: content/developerportal/deploy/managing-local-users-after-deployment.md (hunk 4 above)
Comment (MarkvanMents) [reply content/developerportal/deploy/managing-local-users-after-deployment.md:156]: Ah, OK. So it is basically still a local user but they don't have any roles to sign in to the browser version of the app?

File: content/developerportal/deploy/managing-local-users-after-deployment.md (hunk 5 above)
Comment (MarkvanMents) [reply content/developerportal/deploy/managing-local-users-after-deployment.md:160]: Thanks.

File: content/developerportal/deploy/managing-local-users-after-deployment.md (hunk 2 above)
Comment (MarkvanMents) [reply content/developerportal/deploy/managing-local-users-after-deployment.md:101]: Done

File: content/developerportal/deploy/managing-local-users-after-deployment.md (hunk 3 above)
Comment (MarkvanMents) [reply content/developerportal/deploy/managing-local-users-after-deployment.md]: Changed

File: content/developerportal/deploy/managing-local-users-after-deployment.md (hunk 4 above)
Comment (mengdix) [reply content/developerportal/deploy/managing-local-users-after-deployment.md]: That is my understanding. We can verify it later with the original author(s) of the module.

Comment (MarkvanMents) [reply content/developerportal/deploy/managing-local-users-after-deployment.md]: Have reworded this to reflect your explanation.

**Review by mengdix — APPROVED** [review]

**Review by mengdix — APPROVED** [review]
Good to go!

**Comment by MarkvanMents** [comment]
Thanks for approving Meng (@mengdix)
I have one more question.
Why is the title **Manage Local Users in Studio** when we only talk about local users in Studio **Pro** and the Developer Portal. We don't mention **Studio** at all.

Once I understand that, I can publish it.

**Comment by mengdix** [comment]
Good question, @MarkvanMents . There is no difference between local users in Studio or Studio Pro, since the users are set and managed on the App-level (and not dependent on the modeler). Although, of course, Studio Pro offers app-level user management out of the box. So for Studio, to allow citdevs to create apps and manage users, we include the Administration module so they can add more users /after/ deploying (publishing) the app. 

The use-case is for new Citdev-published apps from Studio on Tencent. Studio *Pro* is only mentioned in this article as a means to import the module, should it not be present yet (for older apps that were made between April and June 2021). Accessing the Administration modules functionalities might not be as apparent for new Citdevs, which is why this help article explains it (it is also linked explicitly in Studio for that purpose). Studio is mentioned several times, especially at the top of the article.

**Comment by MarkvanMents** [comment]
Thanks for the explanation.
I'll leave the title as is and publish it.
Thanks for the collaboration.

**Pull request merged**